crates/sui-types/src/base_types.rs

```diff
@@ -311,6 +312,26 @@ impl From<&Object> for ObjectType {
     }
 }
 
+impl TryFrom<ObjectType> for StructTag {
+    type Error = anyhow::Error;
+
+    fn try_from(o: ObjectType) -> Result<Self, anyhow::Error> {
+        match o {
+            ObjectType::Package => Err(anyhow!("Cannot create StructTag from Package")),
+            ObjectType::Struct(move_object_type) => {
+                let type_params = move_object_type.type_params().to_vec();
+
+                Ok(StructTag {
+                    address: move_object_type.address(),
+                    module: Identifier::new(move_object_type.module().to_string())?,
+                    name: Identifier::new(move_object_type.name().to_string())?,
+                    type_params,
+                })
```

Comment:
```suggestion
                Ok(move_object_type.into())
```

This works because there exists `impl From<MoveObjectType> for StructTag {`